### ISSUE TITLE
Fatal error at start when using a master cli socket with more then one `program` section

### ISSUE THREAD

**Issue by williambailey** [haproxy/haproxy#446]
## Output of `haproxy -vv` and `uname -a`

```
HA-Proxy version 2.0.12 2019/12/21 - https://haproxy.org/
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = /usr/local/bin/gcc
  CFLAGS  = -m64 -I/usr/local/include
  OPTIONS = USE_PCRE=1 USE_PCRE_JIT=1 USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1 USE_NS=

Feature list : +EPOLL -KQUEUE -MY_EPOLL -MY_SPLICE +NETFILTER +PCRE +PCRE_JIT -PCRE2 -PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD -PTHREAD_PSHARED -REGPARM -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H -VSYSCALL +GETADDRINFO +OPENSSL +LUA +FUTEX +ACCEPT4 -MY_ACCEPT4 +ZLIB -SLZ +CPU_AFFINITY +TFO -NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL -SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=4).
Built with OpenSSL version : OpenSSL 1.0.2t  10 Sep 2019
Running on OpenSSL version : OpenSSL 1.0.2t  10 Sep 2019
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : SSLv3 TLSv1.0 TLSv1.1 TLSv1.2
Built with Lua version : Lua 5.3.4
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with PCRE version : 8.42 2018-03-20
Running on PCRE version : 8.42 2018-03-20
PCRE library supports JIT : yes
Encrypted password support via crypt(3): yes

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE        mux=H2
              h2 : mode=HTX        side=FE|BE     mux=H2
       <default> : mode=HTX        side=FE|BE     mux=H1
       <default> : mode=TCP|HTTP   side=FE|BE     mux=PASS

Available services : none

Available filters :
        [SPOE] spoe
        [COMP] compression
        [CACHE] cache
        [TRACE] trace

```

```
Linux host 4.9.184-linuxkit #1 SMP Tue Jul 2 22:58:16 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

## What's the configuration?

```
global
    master-worker

program foo
    command sleep 100

program bar
    command sleep 100
```

## Steps to reproduce the behavior

1. Save the above configuration to `ex.cfg`
2. Start HAProxy with `haproxy -f ex.cfg -S "/tmp/master.sock,mode,600"`

## Actual behavior

HAProxy failes to start.

```
[ALERT] 013/132926 (1378) : parsing [cur--1:0] : proxy 'MASTER', another server named 'cur--1' was already defined at line 0, please use distinct names.
[ALERT] 013/132926 (1378) : Fatal errors found in configuration.
```

## Expected behavior

HAProxy should start without issue.

## Do you have any idea what may have caused this?

This issue only seems to manifest when using the `-S` cli flag.

If not using the `-S` flag everything seems to work as expected.

```
haproxy -f ex.cfg 
[NOTICE] 013/133229 (1382) : New program 'foo' (1383) forked
[NOTICE] 013/133229 (1382) : New program 'bar' (1384) forked
[NOTICE] 013/133229 (1382) : New worker #1 (1385) forked
^C[WARNING] 013/133231 (1382) : Exiting Master process...
[ALERT] 013/133231 (1382) : Current program 'foo' (1383) exited with code 130 (Interrupt)
[ALERT] 013/133231 (1382) : Current program 'bar' (1384) exited with code 130 (Interrupt)
[ALERT] 013/133231 (1382) : Current worker #1 (1385) exited with code 130 (Interrupt)
[WARNING] 013/133231 (1382) : All workers exited. Exiting... (130)
```

## Do you have an idea how to solve the issue?

No, sorry.

**Comment by wlallemand** [comment]
Hello,

It looks like a bind and a server for the interprocess communications are created for the programs, it mustn't since it's not a worker. I'll fix it.

**Comment by wlallemand** [comment]
Fixed and backported to 2.1 and 2.0.
